tests/unit/s2n_cert_info_test.c

```diff
@@ -0,0 +1,139 @@
+ * permissions and limitations under the License.
+ */
+
+#include <errno.h>
+#include <fcntl.h>
+#include <openssl/pem.h>
+#include <openssl/x509.h>
+
+#include "api/s2n.h"
+#include "crypto/s2n_openssl_x509.h"
+#include "s2n_test.h"
+#include "testlib/s2n_testlib.h"
+
+int main(int argc, char **argv)
+{
+    BEGIN_TEST();
+
+    /* s2n_cert_get_cert_info */
+    struct {
+        const char *key_type;
+        const char *signature;
+        const char *key_size;
+        const char *digest;
+        int expected_signature_nid;
+        int expected_digest_nid;
+    } test_cases[] = {
+        { .key_type = "ec",
+                .signature = "ecdsa",
+                .key_size = "p384",
+                .digest = "sha256",
+                .expected_signature_nid = NID_ecdsa_with_SHA256,
+                .expected_digest_nid = NID_sha256 },
+        { .key_type = "ec",
+                .signature = "ecdsa",
+                .key_size = "p256",
+                .digest = "sha384",
+                .expected_signature_nid = NID_ecdsa_with_SHA384,
+                .expected_digest_nid = NID_sha384 },
+        { .key_type = "ec",
+                .signature = "ecdsa",
+                .key_size = "p521",
+                .digest = "sha512",
+                .expected_signature_nid = NID_ecdsa_with_SHA512,
+                .expected_digest_nid = NID_sha512 },
+        { .key_type = "rsae",
+                .signature = "pkcs",
+                .key_size = "2048",
+                .digest = "sha1",
+                .expected_signature_nid = NID_sha1WithRSAEncryption,
+                .expected_digest_nid = NID_sha1 },
+        { .key_type = "rsae",
+                .signature = "pkcs",
+                .key_size = "2048",
+                .digest = "sha224",
+                .expected_signature_nid = NID_sha224WithRSAEncryption,
+                .expected_digest_nid = NID_sha224 },
+        { .key_type = "rsae",
+                .signature = "pkcs",
+                .key_size = "3072",
+                .digest = "sha384",
+                .expected_signature_nid = NID_sha384WithRSAEncryption,
+                .expected_digest_nid = NID_sha384 },
+/* openssl 1.0.* does not support rsapss signatures */
+#if S2N_OPENSSL_VERSION_AT_LEAST(1, 1, 0)
+        { .key_type = "rsae",
+                .signature = "pss",
+                .key_size = "4096",
+                .digest = "sha384",
+                .expected_signature_nid = NID_rsassaPss,
+                .expected_digest_nid = NID_undef },
+        { .key_type = "rsapss",
+                .signature = "pss",
+                .key_size = "2048",
+                .digest = "sha256",
+                .expected_signature_nid = NID_rsassaPss,
+                .expected_digest_nid = NID_undef },
+#endif
+    };
+    for (size_t i = 0; i < s2n_array_len(test_cases); i++) {
+        /* print statement to help debugging in CI */
+        printf("get_cert_info test case %zu\n", i);
```

Comment:
Did you mean to leave in this print statement?

Reply:
The print statement isn't conditional on whether the test fails though. It looks like it's going to print out stuff even if the test succeeds, which we don't want.

Reply:
CMake/CTest captures all stdout by default, and will only show the output when there is a failure in the test (`--output-on-failure`) or if the `--verbose` option is used.